### ISSUE TITLE
How to change the default change_stream_churning_interval?

### ISSUE THREAD

**Issue by jwo-highnote** [GoogleCloudPlatform/cloud-spanner-emulator#190]
Hi, is there a way to change the default change stream interval to say 1 sec? We should like to build regression test on CDC code but the default is 20~40 secs which is too long for CI/CD. I saw there is a change_stream_churning_interval arg but not sure how to pass it to the docker image. Can someone provide an example of the docker run command?

**Comment by skuruppu** [comment]
You're right that we have a flag for that on the C++ side of the code. But it isn't possible to set it when running the Docker image as that runs a Go binary that calls into C++ and the flag also needs to be available in the Go binary.

For example [--overrideMaxDatabasesPerInstance](https://github.com/GoogleCloudPlatform/cloud-spanner-emulator/blob/7f22a69c0d331a980bc6be6c0d3c6117827b16de/binaries/gateway_main.go#L57) is a flag that is defined is Go and is plumbed through to the C++ layer.

I will reach out and see if it's possible to make it configurable through Docker.

**Comment by jwo-highnote** [comment]
Thanks for getting back to me! Also, I think if the default is set to a much smaller time (such as 500ms) is likely what most use case for emulator (testing), IMO. Is this something we can change?

**Comment by skuruppu** [comment]
The team said they chose the 20-40 seconds interval because Emulator is synchronous and will abort and retry any concurrent transactions. A separate thread is used to churn change streams in the background, and this change stream churn transactions will lock the whole database every time they run. So making it too frequent is not recommended (like the 500ms proposed).

Having said that, they said they will look into lowering the default by a few seconds and making it configurable.

We'll keep you posted.

**Comment by skuruppu** [comment]
Just a quick update that we didn't lower the default but we have a change that will make the value configurable through Docker using an environment variable. The change will be in 1.5.26 release which will go out approximately towards the end of the month. Hope this helps.

**Comment by jwo-highnote** [comment]
@skuruppu is there any instruction on how to pass in the parameters? Would like to try the new version as it seems out?

**Comment by jwo-highnote** [comment]
@skuruppu i don't think 1.5.26 has that feature deliver, can you confirm?

**Comment by skuruppu** [comment]
@jwo-highnote I'm so sorry for the late reply. I missed your previous message somehow.

- `--override_change_stream_partition_token_alive_seconds` is the flag introduced in the emulator binary.
- If you're running the Docker image, to overide the flag, you have to set the `CHANGE_STREAM_PARTITION_TOKEN_ALIVE_SECONDS` environment variable to configure this.

Please give this a try and let me know if it works for you.
